.github/workflows/overall-tests.yml

```diff
@@ -12,6 +12,7 @@ jobs:
     uses: ./.github/workflows/partial-tests.yml
 
   coverage:
+    timeout-minutes: 40
```

Comment:
This step runs in // of other tests. there is no point in limiting this to a smaller timeout than the other steps.

I would rollback this.

Reply:
Good point! So I would prefer to put the same timeout as the other steps. I don't like the idea of not having a timeout; we will consume unnecessary energy in case of an infinite loop/proceess stuck. What do you think?

Reply:
The other steps have a timeout. I believe it is 50 minutes. Can you check and align?

Reply:
Other steps are not align in term of timeout. Some 40, some 50

Reply:
Ok, so either we align all the test actions to max 50 minutes, or we accept to have a different time out.

There is nothing blocking to me here. I am ok with both solutions. I let you decide.

Reply:
We can align the timeout in a different PR